### ISSUE TITLE
aria-role="application" should be allowed when a widget role is required.

### ISSUE THREAD

**Issue by gamesaucer** [dequelabs/axe-core#890]
When using an element that is not normally focusable, and making it focusable, a WAI-ARIA widget role is required. However, [the spec states the following in the "application" role:](https://www.w3.org/TR/wai-aria/#application)

> When there is a need to create an element with an interaction model that is not supported by any of the WAI-ARIA `widget` roles, authors _**MAY**_ give that element role `application`.

Therefore, shouldn't it be permitted to use the "application" role when normally a widget role is required? I'm using aXe Coconut 3.3.0.0 and it doesn't recognise this exception. It throws the following violation:

>Elements in the focus order need a role appropriate for interactive content

Here is a code snippet that will produce that violation:

```html
<!DOCTYPE html>
<html lang="en">
    <head>
        <title>Example page</title>
    </head>
    <body><main>
        <h1>Main content</h1>
        <div tabindex="0" role="application">
            Application
        </div>
    </main></body>
</html>
```

**Comment by WilcoFiers** [comment]
Jup, I totally agree. Let me go sort that out. Thanks for reporting!
